flexmeasures/api/v3_0/tests/test_sensor_schedules.py

```diff
@@ -422,3 +422,76 @@ def test_get_schedule_fallback_not_redirect(
         assert schedule["scheduler_info"]["scheduler"] == "StorageFallbackScheduler"
 
         app.config["FLEXMEASURES_FALLBACK_REDIRECT"] = False
+
+
+@pytest.mark.parametrize(
+    "message, flex_config, field, err_msg",
+    [
+        (
+            message_for_trigger_schedule(),
+            "flex-context",
+            "site-consumption-capacity",
+            "no read authorization",
+        ),
+        (
+            message_for_trigger_schedule(),
+            "flex-model",
+            "site-consumption-capacity",
+            "no read authorization",
+        ),
+    ],
+)
+@pytest.mark.parametrize(
+    "requesting_user", ["test_prosumer_user@seita.nl"], indirect=True
+)
+def test_trigger_schedule_with_unauthorized_sensor(
+    app,
+    add_battery_assets,
+    setup_capacity_sensor_on_asset_in_supplier_account,
+    keep_scheduling_queue_empty,
+    message,
+    flex_config,
+    field,
+    err_msg,
+    requesting_user,
+):
+    """Test triggering a schedule using a flex config that refers to a capacity sensor from a different account.
+
+    The user is not authorized to read sensors from the other account,
+    so we expect a 422 (Unprocessable entity) response referring to the relevant flex-config field.
```

Comment:
The user is authorized to call the endpoint, but the payload they send us unprocessible.

Reply:
It's processable, but they are trying to access resources (sensors) they are not authorized to use.

Reply:
403 (Forbidden) would be closer to what you describe and also a closer match to my current implementation.

It's not what I was aiming for, actually, because I was conflating two issues. I wanted the response to suggest that the flex-context is semantically erroneous, because a capacity sensor from a completely different account is used. But you are right that 422 is not a correct response for an auth handler to give. Such a check (for an inapplicable capacity sensor) should be performed within the context of Marshmallow validation. I think it would be better to split the checks in three parts:
- The `auth_required` decorator (which we have already, of course) checks whether the user has the right credentials to hit the endpoint -> 401
- The `permission_required_for_context` decorator checks whether the user has read access to the flex-context sensors -> 403
- A Marshmallow validator checks if the capacity sensors belong to one of the sensor's parent assets (but e.g. the price sensors can come from other assets). -> 422 (follow-up issue)

- https://www.rfc-editor.org/rfc/rfc9110#name-401-unauthorized
- https://www.rfc-editor.org/rfc/rfc9110#name-403-forbidden
- https://www.rfc-editor.org/rfc/rfc9110#name-422-unprocessable-content